spot-client/src/common/ui/loaders/abstract-loader.js

```diff
@@ -18,10 +19,45 @@ export class AbstractLoader extends React.PureComponent {
      *
      * @param {Object} props - The read-only properties with which the new
      * instance is to be initialized.
+     * @param {string} serviceName - The name of the service that will be loaded and which will
+     * appear in the logs.
+     * @param {boolean} supportsReconnects=false - Whether or not the service to be loaded by
+     * the {@code AbstractLoader} subclass supports or is willing to do reconnects.
      */
-    constructor(props) {
+    constructor(props, serviceName = 'abstract-loader', supportsReconnects = false) {
```

Comment:
What's your opinion on calendar and ultrasound also extending abstract loader? Do you think they still should? It's something I can take on to move them off of abstract loader if needed.

Reply:
Yeah, I think they should. At least for now